frontend/packages/ceph-storage-plugin/src/components/ocs-storage-class-form/ocs-storage-class-form.tsx

```diff
@@ -306,57 +311,132 @@ export const StorageClassEncryption: React.FC<ProvisionerProps> = ({
+        UISupportedProviders[providerName]?.allowedPlatforms.includes(infraType)
+      )
+        res.push(
+          <DropdownItem
+            key={providerDetails.group}
+            component="button"
+            id={providerName}
+            data-test={providerDetails.group}
+            onClick={handleProviderDropdownChange}
+          >
+            {providerDetails.group}
+          </DropdownItem>,
+        );
+      return res;
+    },
+    [],
+  );
+
+  React.useEffect(() => {
+    const handleServiceDropdownChange = (e: React.KeyboardEvent<HTMLInputElement>) =>
+      setEncryptionId(e.currentTarget.id);
+    setKmsServiceDropdownItems(
+      _.reduce(
+        csiConfigMap?.data,
+        (res, connectionDetails, connectionName) => {
+          try {
+            // removing any object having syntax error
+            // or, which are not supported by UI.
+            const kmsData = JSON.parse(connectionDetails);
+            if (UISupportedProviders[provider].supported.includes(kmsData?.[KMS_PROVIDER])) {
+              res.push(
+                <DropdownItem
+                  key={connectionName}
+                  component="button"
+                  id={connectionName}
+                  data-test={connectionName}
+                  onClick={handleServiceDropdownChange}
+                  description={kmsData?.[DescriptionKey[kmsData?.[KMS_PROVIDER]]]}
+                >
+                  {connectionName}
+                </DropdownItem>,
+              );
+            }
+          } catch (err) {
+            return err;
+          }
+          return res;
+        },
+        [],
+      ),
     );
-  });
+  }, [provider, csiConfigMap, setEncryptionId]);
 
   return (
-    <div className="form-group">
-      <Dropdown
-        className="dropdown dropdown--full-width"
-        toggle={
-          <DropdownToggle
-            id="kms-dropdown-id"
-            data-test="kms-dropdown-toggle"
-            onToggle={() => setOpen(!isOpen)}
-            toggleIndicator={CaretDownIcon}
-          >
-            {serviceName || t('ceph-storage-plugin~Select an existing connection')}
-          </DropdownToggle>
-        }
-        isOpen={isOpen}
-        dropdownItems={kmsDropdownItems}
-        onSelect={() => setOpen(false)}
-        id="kms-connection"
-        data-test="kms-connection-dropdown"
-      />
+    <div className="ocs-storage-class-encryption__form-dropdown--padding">
+      <div className="form-group">
+        <label htmlFor="kms-provider">{t('ceph-storage-plugin~Provider')}</label>
+        <Dropdown
+          className="dropdown dropdown--full-width"
+          toggle={
+            <DropdownToggle
+              id="kms-provider-dropdown-id"
+              data-test="kms-provider-dropdown-toggle"
+              onToggle={() => setProviderOpen(!isProviderOpen)}
+              toggleIndicator={CaretDownIcon}
+              isDisabled={!isOdf || isLengthUnity(kmsProviderDropdownItems)}
```

Comment:
need to add feature guard here instead of checking for `isOdf`, so that dropdown is disabled for ODF 4.9 + OCP 4.10 cases.